stripe/stripe-ios
# Update Dashboard confirm logic

## Summary
- Add assert if Dashboard uses non-deferred intent when PaymentSheet loads. Non-deferred intent flows aren't supported today because they don't create a PM first, which Dashboard's `uk_` key requires.
- Refactor out unnecessary dashboard helper methods
- Add support for Dashboard SetupIntents: Set moto and make API accommodations for `uk_` to retrieve and confirm SetupIntent API calls, just like https://github.com/stripe-ios/stripe-ios-private/pull/59 

## Motivation
https://jira.corp.stripe.com/browse/RUN_MOBILESDK-2808


## Testing
Added tests (h/t @mludowise-stripe) for Dashboard PI and SI confirm flows that
- Use Dashboard's "id" shaped client secrets
- Check moto is set
- Check client secret is nil (for whatever reason, this is a requirement for uk)

I also manually tested Dashboard's deferred payment and setup flow. 
* PaymentIntent w/ new card: https://admin.corp.stripe.com/payment_intent/pi_3OUc1uL6pqDH2fDJ1DYclWI2
* PaymentIntent w/ saved card: https://admin.corp.stripe.com/payment_intent/pi_3OUcNML6pqDH2fDJ1YQEiKXl
* SetupIntent (using this [branch](https://github.com/stripe-ios/dashboard-ios/tree/stakr/nschris/navIAM1-17)) : https://admin.corp.stripe.com/setup_intent/seti_1OUcMfL6pqDH2fDJ7ATF3bD1

## Changelog
Not user facing
